Update script to support a config file

Rambomst opened the pull request:
- Update script to load key mappings from config files rather than being hard coded.
- Updated readme to support new config changes.
- Added default config json file.

*edit*

I hope these changes are okay, let me know if you want me to make any modifications.

Thanks for creating this original script, I just swapped over to linux from windows and didn't even think to check compatibility with my pedal.

Rambomst committed:
Update script to support a config file.

- Update script to load key mappings from config files rather than being
  hard coded.
- Updated readme to support new config changes.
- Added default config json file.

Rambomst committed:
Update README.md

RoarkGit commented on config.default.json:
```diff
@@ -0,0 +1,16 @@
+{
+    "left_keys": [],
+    "middle_keys": [
+    {
```
nit: These curly brace sections should be indented.

RoarkGit commented on pedal_mapper.py:
```diff
@@ -38,6 +40,14 @@ def __init__(self, left_keys=[], middle_keys=[], right_keys=[], polling_rate=10)
+    def parse_key_combos(self, key_combos):
+        parsed_combos = []
+        for combo in key_combos:
+            mods = [getattr(e, mod) for mod in combo["mods"]]
```
It would be convenient if "mods" could be optional; I tried testing by adding a key to the config that didn't have "mods" set and it didn't run.

RoarkGit commented on README.md:
```diff
@@ -24,6 +24,62 @@ or equivalent to have it always running in the background.
+{
+    "left_keys": [],
+    "middle_keys": [
+    {
```
nit: This should be indented.

RoarkGit reviewed (changes requested):
Overall looks good, just some minor nitpicky things. Thanks for the contribution!

Rambomst committed:
- Updated tabbing for json example.
- Updated parse_key_combos to allow the config key option "mods" to be
  optional.

Rambomst commented:
@RoarkGit All of those issues should be resolved, let me know if there is anything else you would like changed/added.

:)

Rambomst committed:
- Updated README to include instructions on how to create a systemd
  service

Rambomst commented:
I updated the README to include more systemd information

RoarkGit commented:
> I updated the README to include more systemd information

Thanks, I was going to do that as well but you beat me to it!
